spring-cloud-deployer-resource-support/src/main/java/org/springframework/cloud/deployer/resource/registry/UriRegistryPopulator.java

```diff
@@ -73,6 +72,15 @@ public void setResourceLoader(ResourceLoader resourceLoader) {
 				for (String key : properties.stringPropertyNames()) {
 					try {
 						URI uri = new URI(properties.getProperty(key));
+						if (!overwrite) {
+							try  {
+								registry.find(key);
+							}
+							catch (IllegalArgumentException e) {
+								// this key already exists
```

Comment:
thanks @ilayaperumalg - good catch! your changes look good except one minor detail: *exist* (no s at the end) in the catch block comment